pkg/tekton/pipeline_info.go

```diff
@@ -244,14 +245,39 @@ func CreatePipelineRunInfo(prName string, podList *corev1.PodList, ps *v1.Pipeli
 				}
 			}
 		}
-		if lastCommitSha == "" && pullPullSha != "" {
-			lastCommitSha = pullPullSha
+	}
+
+	if pullBaseSha != "" {
+		pri.BaseSHA = pullBaseSha
+	}
+
+	if pullPullSha != "" {
```

Comment:
So I think I knew that once, but what is a pullPullSha again?

Reply:
The `PULL_PULL_SHA` env var is set for pull request builds (hence the first `PULL`) with the PR's `HEAD` SHA (hence the `PULL_SHA` part).

The other relevant env vars are the following - the first three are always set, even for non-PR builds, which is confusing but hey.
* `PULL_BASE_REF` - i.e., `master`
* `PULL_BASE_SHA` - the SHA for master's `HEAD`.
* `PULL_REFS` - basically a comma-separated list of `(branch):(sha)`, with `master` always first and then the PR(s) as `1:(sha),2:(other sha),...`

And then just on PRs:
* `PULL_NUMBER` - which kinda speaks for itself.